packages/cli-repl/src/cli-repl.ts

```diff
@@ -413,6 +420,11 @@ export class CliRepl implements MongoshIOProvider {
         commandLineLoadFiles,
         evalScripts
       );
+
+      this.bus.emit('mongosh:start-session', {
```

Comment:
just a question - why are we emitting `start-session` after evaluating the user's scripts in non-interactive mode?  It seems like when executing scripts, the session has been started after all dependencies have been loaded but before executing any scripts.

Reply:
Ah, you are right, totally right. start-session should be emitted just before `mongosh:start-loading-cli-scripts`.

Thanks a lot for noticing! I just missed this one.